hazelcast/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -450,6 +450,18 @@ void deleteJob(long jobId) {
      * Cleans up stale maps related to jobs
      */
     void cleanup(NodeEngine nodeEngine) {
+        if (!jobRecordsMapExists()) {
```

Comment:
`cleanup` is invoked from a schedule that is created in `DefaultNodeExtension.afterStart()` which in EE is invoked after `hotRestartService.start()` finished. After `hotRestartService.start()` returns, `isStartCompleted()` is always true.
(yes, this is quite convoluted)

Reply:
this check here prevents us from doing cleanup if the IMap does not exists - we may not yet know that it exists. So we are on the safe side (I hope that when IMap exists, it will contain all data...), but miss some opportunities for cleanup with is not a big problem.